RFCXXX-Native-Json-Adapters.md

```diff
@@ -0,0 +1,258 @@
+some parameters must be in a specific location in the command line.
+For example, there are certain parameters which apply to the executable and not the subcommand:
+
+```powershell
+PS> docker --debug run --rm -it mcr.microsoft.com/powershell:latest 
+```
+
+The use of `--debug` has a required position, it _must_ provided before the `run` subcommand.
+
+An addition complication is that depending on the application, parameters should _not_ be added based on the earlier parts of the command.
+For example, you may want to provide different formatting for `git log` vs `git branch`, so you must have more information to determine the proper behavior.
+This means that rather than just capturing the parameter name and default value, we will need to be able to run arbitrary code (ala scriptblock)
+which includes the command line AST so the proper parameter values may be created.
+
+However, with an approach which provides for this complexity, we can continue to simplify the experience for our users to from:
+
+```powershell
+PS> docker --debug run --rm -it mcr.microsoft.com/powershell
+```
+
+to
+
+```powershell
+PS> docker run mcr.microsoft.com/powershell
+```
+
+or
+
+```powershell
+PS> Docker image ls --format='{{json .}}' | ConvertFrom-Json | Where-Object {$_.Repository -match "mcr"}| Format-Table Id,Repository,Tag,Size,CreatedAt
+
+ID           Repository                             Tag                     Size   CreatedAt
+--           ----------                             ---                     ----   ---------
+4e9617ada198 mcr.microsoft.com/powershell           preview-7.3-mariner-2.0 235MB  2022-09-01 11:48:17 -0700 PDT
+. . .
+```
+
+to this:
+
+```powershell
+PS> docker image ls | ? Repository -match "mcr"
+
+ID           Repository                             Tag                     Size   CreatedAt
+--           ----------                             ---                     ----   ---------
+4e9617ada198 mcr.microsoft.com/powershell           preview-7.3-mariner-2.0 235MB  2022-09-01 11:48:17 -0700 PDT
+. . .
+```
+
+### Follow-on Work
+
+A number of follow-on activities should be considered to bring continued improvement of the native command experience.
+
+#### Improved TabCompletion
+
+Current tab-completion (on Unix) is done via a somewhat fragile call to the native shell.
+Tab-completion on Windows must be implemented specifically.
+we should take the opportunity to improve our tooling around tab-completion to make it easier to create,
+and provide generalized help parsers (where possible) to generate tab-completion.
+`bash` provides a number of more generalized tab-completion, we should attempt to do the same for PowerShell users.
+
+For example, bash has a single completer all known hosts which is used by nearly 20 executables.
+We already will tab-complete for file system items,
+a small effort should be able to increase our coverage for other native executables.
+
+#### Improved Formatting Tooling
```

Comment:
For formatting, do `TypeNames` get automatically added to the resulting object?  Maybe something like: `<CommandName>.JsonAdapted` so formatting can be built for it?